src/dmd/dtemplate.d

```diff
@@ -8266,3 +8281,29 @@ void printTemplateStats()
         printf("%8u %8u   %s\n", ts.numInstantiations, ts.uniqueInstantiations, (cast(const TemplateDeclaration) td).toChars());
     }
 }
+
+struct Pool(T)
+if (is(T == class))
+{
+    private static T root;
+
+    static T make(A...)(auto ref A args)
+    {
+        if (!root)
+            return new T(args);
+        auto result = root;
+        root = *(cast(T*) root);
+        memcpy(cast(void*) result, T.classinfo.initializer.ptr, T.classinfo.initializer.length);
+        result.__ctor(args);
+        return result;
+    }
+
+    static void dispose(T goner)
+    {
+        // Front insert into the freelist. "Next" pointer is in the first bytes of the former object.
+        *(cast(T*) goner) = root;
+        root = goner;
+        // Obliterate the memory so attempting to reuse the object after dispose is likely to fail
+        memset(cast(ubyte*) goner + (void*).sizeof, 0xf0, T.classinfo.initializer.length - (void*).sizeof);
```

Comment:
is't a problem here would be that **pointer to** the disposed instances will think their target is still alive ?

Reply:
That's actually good - we don't want the GC to collect objects from the freelist.